scripts/cleanup_edomain_templates.py

```diff
@@ -0,0 +1,252 @@
+        template_creation_time = template.get_creation_time() + delta
+
+        if template.get_name().startswith('auto-tmp'):
+            if now > (template_creation_time + delta).isoformat():
+                delete_templates.append(template)
+
+    if len(delete_templates) == 0:
+        print "RHEVM: No old templates to delete in {}".format(edomain)
+
+    for delete_template in delete_templates[:max_templates]:
+        thread = Thread(target=delete_edomain_templates,
+                        args=(api, delete_template, edomain))
+        thread.daemon = True
+        thread_queue.append(thread)
+        thread.start()
+
+    for thread in thread_queue:
+        thread.join()
+
+
+def api_params_resolution(item_list, item_name, item_param):
+    """Picks and prints info about parameter obtained by api call.
+
+    Args:
+        item_list: List of possible candidates to pick from.
+        item_name: Name of parameter obtained by api call.
+        item_param: Name of parameter representing data in the script.
+    """
+    if len(item_list) == 0:
+        print "RHEVM: Cannot find {} ({}) automatically.".format(item_name, item_param)
+        print "Please specify it by cmd-line parameter '--{}' or in cfme_data.".format(item_param)
+        return None
+    elif len(item_list) > 1:
+        print "RHEVM: Found multiple instances of {}. Picking '{}'.".format(item_name, item_list[0])
+    else:
+        print "RHEVM: Found {} '{}'.".format(item_name, item_list[0])
+
+    return item_list[0]
+
+
+def get_edomain(api):
+    """Discovers suitable export domain automatically.
+
+    Args:
+        api: API to RHEVM instance.
+    """
+    edomain_names = []
+
+    for domain in api.storagedomains.list(status=None):
+        if domain.get_type() == 'export':
+            edomain_names.append(domain.get_name())
+
+    return api_params_resolution(edomain_names, 'export domain', 'edomain')
+
+
+def make_kwargs(args, cfme_data, **kwargs):
+    """Assembles all the parameters in case of running as a standalone script.
+       Makes sure, that the parameters given by command-line arguments
+       have higher priority.Makes sure, that all the needed parameters
+       have proper values.
+
+    Args:
+        args: Arguments given from command-line.
+        cfme_data: Data in cfme_data.yaml
+        kwargs: Kwargs generated from
+        cfme_data['template_upload']['template_upload_rhevm']
+    """
+    args_kwargs = dict(args._get_kwargs())
+
+    if not kwargs:
+        return args_kwargs
+
+    template_name = kwargs.get('template_name', None)
+    if template_name is None:
+        template_name = cfme_data['basic_info']['appliance_template']
+        kwargs.update(template_name=template_name)
+
+    for kkey, kval in kwargs.iteritems():
+        for akey, aval in args_kwargs.iteritems():
+            if aval:
+                if kkey == akey:
+                    if kval != aval:
+                        kwargs[akey] = aval
```

Comment:
Oh, did not notice this thing before, bad me :)

1) The condition can be written in one line, saving the spaghetti (`if aval and kkey == akey and kval != aval:`)
2) You are possibly changing `kwargs` while iterating over it with an iterator. Using `.items()` instead of `.iteritems()` should be better in this case. Unless we switch to python3 which does not have iteritems and its items equals py2 iteritems.